app/parse.py

```
@@ -14,10 +22,95 @@ class Product:
     rating: int
     num_of_reviews: int
 
+def scrape_products_from_page(driver: webdriver.Chrome, url: str, product_selector: str, use_more_button: bool = False) -> List[Product]:
+    driver.get(url)
+    wait = WebDriverWait(driver, 10)
+
+    try:
+        accept_btn = wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, ".accept-cookies-button")))
+        accept_btn.click()
+    except:
+        pass
+
+    products: List[Product] = []
+
+    if use_more_button:
+        while True:
+            try:
+                more_btn = driver.find_element(By.CSS_SELECTOR, ".btn.btn-primary.btn-lg.btn-block")
+                if more_btn.is_displayed():
+                    driver.execute_script("arguments[0].click();", more_btn)
+                    time.sleep(1)
+                else:
+                    break
+            except:
+                break
+
+    items = driver.find_elements(By.CSS_SELECTOR, product_selector)
+    for item in items:
+        try:
+            title = item.find_element(By.CSS_SELECTOR, "a.title").get_attribute("title")
+        except:
+            title = ""
+        try:
+            description = item.find_element(By.CSS_SELECTOR, ".description").text
+        except:
+            description = ""
+        try:
+            price = float(item.find_element(By.CSS_SELECTOR, ".price").text.replace("$", ""))
+        except:
+            price = 0.0
+        try:
+            stars = item.find_elements(By.CSS_SELECTOR, ".ratings .ws-icon-star")
+            rating = len(stars)
+        except:
+            rating = 0
+        try:
+            num_reviews_text = item.find_element(By.CSS_SELECTOR, ".ratings .pull-right, .review-count span").text
+            num_of_reviews = int(num_reviews_text.split()[0])
+        except:
+            num_of_reviews = 0
```

Multiple `try/except` blocks in the per-item parsing use bare `except:` which swallows all exceptions and may hide bugs (selector typos, conversion errors). Replace bare `except` with specific exceptions (`NoSuchElementException`, `ValueError`, etc.) and consider logging unexpected exceptions so parsing issues are visible during tests and debugging.